src/relayserver/penalizer/PenalizerService.ts

```diff
@@ -0,0 +1,231 @@
+// @ts-ignore
+import abiDecoder from 'abi-decoder'
+import { PrefixedHexString, Transaction as EthereumJsTransaction, TransactionOptions, TxData } from 'ethereumjs-tx'
+import { bufferToHex, bufferToInt, isZeroAddress } from 'ethereumjs-util'
+import { Transaction as Web3CoreTransaction } from 'web3-core'
+
+import RelayHubABI from '../../common/interfaces/IRelayHub.json'
+import PayMasterABI from '../../common/interfaces/IPaymaster.json'
+import StakeManagerABI from '../../common/interfaces/IStakeManager.json'
+
+import ContractInteractor from '../../relayclient/ContractInteractor'
+import VersionsManager from '../../common/VersionsManager'
+import { BlockExplorerInterface } from './BlockExplorerInterface'
+import { getDataAndSignature } from '../../common/Utils'
+import replaceErrors from '../../common/ErrorReplacerJSON'
+import { TransactionManager } from '../TransactionManager'
+import { ServerAction } from '../StoredTransaction'
+import { LoggerInterface } from '../../common/LoggerInterface'
+import { gsnRuntimeVersion } from '../../common/Version'
+
+abiDecoder.addABI(RelayHubABI)
+abiDecoder.addABI(PayMasterABI)
+abiDecoder.addABI(StakeManagerABI)
+
+export enum Accusations {
+  repeatedNonce = 'repeatedNonce',
+  illegalTransaction = 'illegalTransaction'
+}
+
+export interface PenalizeRequest {
+  signedTx: PrefixedHexString
+}
+
+export interface PenalizerParams {
+  transactionManager: TransactionManager
+  contractInteractor: ContractInteractor
+  txByNonceService: BlockExplorerInterface
+  devMode: boolean
+}
+
+// TODO: parseInt is dangerous here, convert directly to buffer
+function createWeb3Transaction (transaction: Web3CoreTransaction, rawTxOptions: TransactionOptions): EthereumJsTransaction {
+  const txData: TxData = {
+    gasLimit: transaction.gas,
+    gasPrice: parseInt(transaction.gasPrice),
+    to: transaction.to ?? '',
+    data: transaction.input,
+    nonce: transaction.nonce,
+    value: parseInt(transaction.value),
```

Comment:
for gasprice, parseInt is OK, but for value, it isn't!

Reply:
Will fix